diff --git a/python/sglang/srt/models/qwen3_5.py b/python/sglang/srt/models/qwen3_5.py
index 2268b835d9..7d70a93bbf 100644
--- a/python/sglang/srt/models/qwen3_5.py
+++ b/python/sglang/srt/models/qwen3_5.py
@@ -137,8 +137,8 @@ _qknorm_use_alt_stream = _is_cuda or (
     get_bool_env_var("SGLANG_QK_NORM_ALT_STREAM", "False") and _hip_use_alt_stream
 )
 _gdn_decode_fused_proj_conv = (
-    _is_cuda and envs.SGLANG_ENABLE_GDN_DECODE_FUSED_PROJ_CONV.get()
-)
+    _is_cuda or _is_hip
+) and envs.SGLANG_ENABLE_GDN_DECODE_FUSED_PROJ_CONV.get()
 _is_amx_available = cpu_has_amx_support()
 _is_xpu = is_xpu()
 
